spring-integration-core/src/main/java/org/springframework/integration/channel/MessagePublishingErrorHandler.java

```diff
@@ -140,8 +162,12 @@ public final void handleError(Throwable t) {
 	}
 
 	private MessageChannel resolveErrorChannel(Throwable t) {
-		Message<?> failedMessage = (t instanceof MessagingException) ?
-				((MessagingException) t).getFailedMessage() : null;
+		Throwable actualThrowable = t;
+		if (t instanceof OriginalMessageContainingMessagingException) {
+			actualThrowable = t.getCause();
+		}
+		Message<?> failedMessage = (actualThrowable instanceof MessagingException) ?
```

Comment:
According to the `OriginalMessageContainingMessagingException` ctor the cause is always `MessagingException`.

Do you have anything else in mind?